mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,289 @@
+        a display area of 531 x 298 mm.
+    screen_distance : float
+        The distance (in meters) from the participant's eyes to the screen.
+    screen_resolution : array-like of shape ``(2,)``
+        The resolution (in pixels) of the screen that the eyetracking data
+        was collected with. For example, ``(1920, 1080)`` for a 1920x1080
+        resolution display.
+
+    Attributes
+    ----------
+    onset : float
+        The onset of the calibration in seconds. If the calibration was
+        performed before the recording started, the onset will be ``0`` seconds.
+    model : str
+        A string, which is the model of the calibration that was applied. For
+        example ``'H3'`` for a horizontal only 3-point calibration, or ``'HV3'`` for a
+        horizontal and vertical 3-point calibration.
+    eye : str
+        The eye that was calibrated. For example, ``'left'``, or ``'right'``.
+    avg_error : float
+        The average error in degrees between the calibration points and the actual gaze
+        position.
+    max_error : float
+        The maximum error in degrees that occurred between the calibration points and
+        the actual gaze position.
+    positions : ndarray of float, shape ``(n_calibration_points, 2)``
+        The x and y coordinates of the calibration points.
+    offsets : ndarray of float, shape ``(n_calibration_points,)``
+        The error in degrees between the calibration position and the actual
+        gaze position for each calibration point.
+    gaze : ndarray of float, shape ``(n_calibration_points, 2)``
+        The x and y coordinates of the actual gaze position for each calibration point.
+    screen_size : array-like
+        The width and height (in meters) of the screen that the eyetracking data was
+        collected  with. For example ``(.531, .298)`` for a monitor with a display area
+        of 531 x 298 mm.
+    screen_distance : float
+        The distance (in meters) from the participant's eyes to the screen.
+    screen_resolution : array-like
+        The resolution (in pixels) of the screen that the eyetracking data was
+        collected with. For example, ``(1920, 1080)`` for a 1920x1080 resolution
+        display.
+    """
+
+    def __init__(
+        self,
+        onset=None,
+        model=None,
+        eye=None,
+        avg_error=None,
+        max_error=None,
+        positions=None,
+        offsets=None,
+        gaze=None,
+        screen_size=None,
+        screen_distance=None,
+        screen_resolution=None,
+    ):
+        super().__init__(
+            onset=onset,
+            model=model,
+            eye=eye,
+            avg_error=avg_error,
+            max_error=max_error,
+            screen_size=screen_size,
+            screen_distance=screen_distance,
+            screen_resolution=screen_resolution,
+        )
+        self["positions"] = positions
+        self["offsets"] = offsets
+        self["gaze"] = gaze
```

Comment:
The reason is that the ``_set_item__`` dunder method (that I just added since your last review), checks if the values of these keys are `array-like`, and  if so, converts them to numpy arrays. But if you pass a value to these keys directly in the constructor ``__set_item__`` isn't called (at least when I tested it). 

Am I over complicating things? The idea was that the ``plot`` method expects those keys to be numpy arrays so I was try to make sure they are during the instance construction.

Reply:
> Am I over complicating things? The idea was that the plot method expects those keys to be numpy arrays so I was trying to make sure they are during the instance construction.

For simplicity you could probably assume that no user would/should ever set these, so you can assume they are all `ndarray` of the right shape. You don't currently check for example that they are all shape `(n, 2)` or `(n,)`, so the check is currently incomplete anyway. So maybe one way to go is to put a `.. note::` in the class docstring saying that this class is not meant to be constructed by the user.

Another -- if you want to be more complete -- option is to add a `self._check_consistency` method that checks and sets (if necessary) the values to be `ndarray` of the right shape, checking to make sure the shapes are okay. That way you can avoid overriding the `__setitem__` and just call `self._check_consistency()` at the end of `__init__` and also maybe at the start of `self.plot`.

Reply:
> So maybe one way to go is to put a `.. note::` in the class docstring saying that this class is not meant to be constructed by the user.

Yeah, I already added a `.. note::` telling users that when possible the calibration instance should be created with a helper function.

So in that case and considering your point that my check was not complete anyway, I think I'll remove the custom ``__set_item__`` dunder method, and we can get a lay of the land after this PR is shipped;  if  it ends up that a lot of people need to create `Calibration` instances from scratch and are running into issues then we can create a robust `_check_consistency` routine?